cobigen-templates/templates-devon4j/src/main/templates/crud_java_server_app/templates.xml

```diff
@@ -57,6 +61,10 @@
       <templateRef ref="${variables.entityName}Entity.java"/>
       <templateRef ref="${variables.entityName}.java"/>
     </increment>
+    <increment name="dataaccess_infrastructure_composedkey" description="Entity infrastructure (ComposedKey)">
+      <templateRef ref="ApplicationComposedKeyPersistenceEntity.java"/>
+      <templateRef ref="ApplicationComposedKeyEntity.java"  />
+    </increment>
```

Comment:
We need to discuss what this increments are about.
For the user, it seems, that he can generate `Entity Infrastructure (Composed Key)`, but actually this does nothing, as it just generates the abstract classes. So you always need to select `Entity infrastructure` in addition to make it work.
This is a little strange and cannot be explained easily to the user.

The problem here is, that we are facing an insufficiency of CobiGen to conditionally generate templates on demand (e.g. if a type is generated, which does not yet exist). Although, this might be an interesting feature, my next question would be, why isn't there support in devon4j for composite keys? Has this discussion being raised at devon4j repository?

@jdiazgon, @sjimenez77, @hohwille ?

Reply:
I would strongly encourage to avoid creating additional increments for this.
IMHO we have an entity as input and can already generate sufficient output for this.
The only problems are:
1. The name of the Entity root class is currently "hardwired" as `ApplicationPersistenceEntity` in the templates and needs more flexibility
2. The ID used by some CRUD use-cases is currently "hardwired" as `Long` in the templates and needs more flexibility

For 1. my suggestion is to use a pattern instead `Application*Entity` as we generally use the `Application` prefix for base-classes (e.g. shipped with the app-template). So this would IMHO make sense.

For 2. my suggestion would be to determine the type of the ID via reflection. We can IMHO make the assumtion or more precisely the requirement that the property name has to be "id" so via the return type of `getId()` Method we could figure out the type of the ID. Instead of hardcoding `Long` we would instead then need to use the ID type we determined via reflection.
Projects then need to configure Jackson mappings, etc. to make their custom ID properly supported from REST service to Database and back.